File: backend/app/schemas/trade.py
from pydantic import BaseModel
from decimal import Decimal
from datetime import datetime


class TradeCreate(BaseModel):
    symbol: str
    side: str
    quantity: Decimal
    price: Decimal
    fee: Decimal = Decimal("0")
    slippage: Decimal = Decimal("0")
    timestamp: datetime

File: frontend/src/types/trade.ts
export interface Trade {
	trade_id: number;
	portfolio_id: number;
	symbol: string;
	side: 'BUY' | 'SELL';
	quantity: string;
	price: string;
	fee: string;
	slippage: string;
	realized_pnl: string;
	timestamp: string;
}
